Automattic/woocommerce-payments
# Fix currency formatting for totals on My Account > Subscriptions page

Fixes #5488

#### Changes proposed in this Pull Request

* Remove `static` declaration for `MultiCurrency->is_initialized()` method.
* Add `current_my_account_subscription` property to hold the current subscription being iterated through on the My Account > Subscriptions page.
* Add `frontend_currencies` property for the `FrontendCurrencies` class/object to be used in the class.
* Add three new filters. Two filters are used as trigger events, and then the `wc_price` filter will add the 3 letter currency code, if needed.
* Add another check in `override_selected_currency` method to check if `current_my_account_subscription` is set, and if so, use that subscription's currency.
* Add `maybe_set_current_my_account_subscription` method that is called by `woocommerce_subscription_price_string_details` filter. This will set the value of `current_my_account_subscription` to the passed subscription if the backtrace calls are found, and it does not actually filter any values.
* Add `maybe_clear_current_my_account_subscription` method that is called by `woocommerce_get_formatted_subscription_total` filter. This will set the value of `current_my_account_subscription` to `null` if the backtrace calls are found, and it does not actually filter any values.
* Add `maybe_get_explicit_format_for_subscription_total` method that checks for a subscription in `current_my_account_subscription`, and if found it will use `WC_Payments_Explicit_Price_Formatter::get_explicit_price_with_currency` to add the three letter currency code to the price string.
* Add `is_current_my_account_subscription_set` method to check to see if `current_my_account_subscription` is a subscription.

I added a note regarding a test that is currently marked as skipped due to I could not find a way to make it work.

#### Testing instructions

> [!Note]
> This requires the Woo Subscriptions plugin to be installed.
> There is a bug with rounding if a zero decimal currency is selected: https://github.com/Automattic/woocommerce-payments/issues/7537 

* Do not check out this branch yet.
* If you do not have subscriptions active with different currencies on your test site, you will need to create them.
  * If needed: Navigate to WooCommerce > Settings > Multi-Currency and add other currencies, such as: EUR, GBP, JPY.
  * If needed: Navigate to Products > Add New and add a new subscription for 100 a day.
  * If needed: Navigate to your subscription on the front end and purchase in multiple currencies.
* Navigate to My Account > Subscriptions.
* Change the currency to JPY and you will see the formatting of the totals for the currencies change to the JPY format.
* Check out this branch, then refresh the page.
* Change the currency to multiple different currencies, the format for the total amounts should remain proper for the currency itself and should not take on the formatting for the currency selected.

-------------------


**Post merge**
